File: blackjack_card_counter.java

import java.util.Scanner;
import java.util.HashMap;

class blackjack_card_counter {

    int getStats(int current_player_count, ){
        //Declare variables that will be used to calculate statistics
        int totalCards,cardUnder21,cardEqual21,cardOver21;
        totalCards = 0;
        cardUnder21 = 0;
        cardEqual21 = 0;
        cardOver21 = 0;
        //loop through the count of each card and increment variables
        for (Map.Entry mapElement : cardCount.entrySet()) {
            int currentCardValue = (int)mapElement.getKey();
            int cardCardCount = ((int)mapElement.getValue());

            totalCards += currentCardCount;

            if (currentCardValue < 21){
                cardUnder21 += currentCardCount;
            }else if(currentCardValue > 21){
                cardOver21 += currentCardCount;
            }else{
                cardEqual21 += currentCardCount;
            }
        //print the stats to the user
        System.out.println("Under count: ");
        System.out.println("Equal count: ");
        System.out.println("Over count: ");
        return 0;
    }
    }

    boolean updateDeck(int currentPlayerCount, int nextCard){
        if (cardCount.get(nextCard) == 1){
            return false;
        }else{
            if (currentPlayerCount + nextCard > 21){
                return false;
            else{
                cardCount.put(nextCard,cardCount.get(nextCard)-1);
                return true;
            }
            }
        }
    }

    public static void main(String[] args) {
        public static HashMap<Integer, Integer>cardCount = new HashMap<Integer, Integer>();
        boolean GAME_ON = true;
        int player_count = 0;
        int dealer_count = 0;
        //populate the hashmap
        for (int card = 1; card < 12; card++){
            cardCount.put(card,4);
        }

        while (GAME_ON){

        }

        Scanner myObj = new Scanner(System.in);
        System.out.println("Enter name, age and salary:");

    // String input
        String name = myObj.nextLine();

    // Numerical input
        int age = myObj.nextInt();
        double salary = myObj.nextDouble();

    // Output input by user
        System.out.println("Name: " + name);
        System.out.println("Age: " + age);
        System.out.println("Salary: " + salary);
    }
}

File: blackjack_card_counter.py
def get_stats(player_total,card_count):
  total_cards = 0
  card_under_21 = 0
  card_equal_21 = 0
  card_over_21 = 0
  
  for current_card in card_count:
    total_cards += card_count[current_card]
    if player_total + current_card < 21:
      card_under_21 += card_count[current_card]
    elif player_total + current_card > 21:
      card_over_21 += card_count[current_card]
    else:
      card_equal_21 += card_count[current_card]

  #print the stats
  print ("Under count: {}".format(card_under_21/total_cards))
  print ("Exact count: {}".format(card_equal_21/total_cards))
  print ("Over count: {} \n".format(card_over_21/total_cards))
  

def update_deck(player_count,next_card):
  global card_count
  if card_count[next_card] == 1:
    return False
  else:
    if player_count + next_card > 21:
      return False
    card_count[next_card] -= 1
    return True

GAME_ON = True
card_count = {}

for i in range(1,12):
  card_count[i] = 4
  
player_count = 0
dealer_count = 0

while GAME_ON:  
  player_next = input("Next card for player {}: ".format(player_count))
  player_count += int(player_next)
  GAME_ON = update_deck(player_count,int(player_next))
  get_stats(player_count,card_count)
  
  dealer_next = input("Next card for dealer {}: ".format(dealer_count))
  dealer_count += int(dealer_next)
  GAME_ON = update_deck(dealer_count,int(dealer_next))
  get_stats(dealer_count,card_count)
  
  
  
